share/p2p/discovery/manager.go

```diff
@@ -0,0 +1,51 @@
+package discovery
+
+import (
+	"context"
+	"errors"
+)
+
+type Manager struct {
+	discs []*Discovery
+
+	cancel context.CancelFunc
+}
+
+func NewManager(discs []*Discovery) *Manager {
+	return &Manager{
+		discs: discs,
+	}
+}
+
+func (m *Manager) Start(context.Context) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	m.cancel = cancel
+
+	for _, d := range m.discs {
+		err := d.Start(ctx)
+		if err != nil {
+			log.Errorw("failed to start discovery", "err", err)
```

Comment:
If there are multiple discoveries and last one failed the function will return. Perhaps started ones should be stopped before exit.

Reply:
this could be resolved if fx called Stop if Start partially fails on the node

Reply:
https://github.com/celestiaorg/celestia-node/issues/3408

Reply:
Likely unneeded with approach described in https://github.com/celestiaorg/celestia-node/pull/3188#discussion_r1604349569